File: Java/Gestor_canciones/src/baseDatos/Deserializador.java
package baseDatos;

import java.io.EOFException;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.util.ArrayList;
import java.util.List;

import gestorAplicacion.elementosLibreria.Cancion;
import gestorAplicacion.elementosLibreria.Biblioteca;
import gestorAplicacion.elementosLibreria.Favoritos;
import gestorAplicacion.elementosLibreria.Me_gusta;
import gestorAplicacion.elementosLibreria.Mis_artistas;
import gestorAplicacion.elementosLibreria.Playlist;
import gestorAplicacion.personas.Artista;
import gestorAplicacion.personas.Usuario;



public class Deserializador {
	public static <E> void deserializador(List<E> list, String className) {
		FileInputStream fileIn;
		try {
			// Creamos una cadena con la ruta del archivo que vamos a cargar
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + className + ".txt";
			System.out.println(path);
			// Utilizamos un file para crear este archivo si no existe aun
			File archivo = new File(path);
			archivo.createNewFile(); // Crea un nuevo archivo si no existe

			// Usamos un FileInputStream para poder saber de donde cargar el archivo
			fileIn = new FileInputStream(path);

			// Si el archivo esta vacio se lanza un throw EOFException y se muestra como un mensaje de vacio, pero si no se usa el objeto in para leer el archivo
			ObjectInputStream in = new ObjectInputStream(fileIn);

			// Lee el listado de elementos
			ArrayList<E> listado = (ArrayList<E>) in.readObject();

			// Recorro el ArrayList
			for (E el : listado) {
				list.add(el);
			}

			in.close();
			fileIn.close();

		} catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			System.out.println("Esta vacio");
		} catch (ClassNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}

	/* Funcion para deserializar toda la aplicacion Generic IT  */
	public static void deserializarTodo() {
		Deserializador.deserializador(Biblioteca.getBibliotecas() , "Biblioteca");
		Deserializador.deserializador(Cancion.getCancionesCreadas(), "Canciones");
		Deserializador.deserializador(Favoritos.getFavoritos() , "Favotitos");
		Deserializador.deserializador(Me_gusta.getMe_gustas() , "Me gusta");
		Deserializador.deserializador(Mis_artistas.getMis_artistas() , "Mis artistas");
		Deserializador.deserializador(Playlist.getPlaylists(), "Playlists");
		Deserializador.deserializador(Artista.getArtistas(), "Artistas");
		Deserializador.deserializador(Usuario.getUsuarios(), "Usuarios");
	}
}

File: Phyton/Gestor_canciones/src/baseDatos/Deserializador.py
import pickle
from gestorAplicacion.elementosLibreria.Cancion import Cancion
from gestorAplicacion.elementosLibreria.Biblioteca import Biblioteca
from gestorAplicacion.elementosLibreria.Favoritos import Favoritos
from gestorAplicacion.elementosLibreria.Me_gusta import Me_gusta
from gestorAplicacion.elementosLibreria.Mis_artistas import Mis_artistas
from gestorAplicacion.elementosLibreria.Playlist import Playlist
from gestorAplicacion.personas.Artista import Artista
from gestorAplicacion.personas.Usuario import Usuario
import pathlib
import os

class Deserializador():
    
    def deserializar(lista, className):
        def camino(className):
            string = os.path.join(pathlib.Path(__file__).parent.absolute(), "temp\\"+className+".txt")
            return string
        # Leo el archivo
        try:
            picklefile = open(camino(className), 'rb')
        except:
            picklefile = open(camino(className), 'x')
            picklefile = open(camino(className), 'rb')
        # unpickle los datos
        if os.path.getsize(camino(className)) > 0:
            lista = pickle.load(picklefile)
        
        # Cierro el archivo
        picklefile.close()
        return lista
        # Cierro el archivo
    
    def deserializarTodo():
        Biblioteca.bibliotecas = Deserializador.deserializar(Biblioteca.bibliotecas, "Biblioteca")
        Cancion.cancionesCreadas =  Deserializador.deserializar(Cancion.cancionesCreadas, "Canciones")
        Favoritos.favoritos = Deserializador.deserializar(Favoritos.favoritos, "Favotitos")
        Me_gusta.me_gustas = Deserializador.deserializar(Me_gusta.me_gustas, "Me gusta")
        Mis_artistas.mis_artistas = Deserializador.deserializar(Mis_artistas.mis_artistas, "Mis artistas")
        Playlist.playlists = Deserializador.deserializar(Playlist.playlists, "Playlists")
        Artista.artistas = Deserializador.deserializar(Artista.artistas, "Artistas")
        Usuario.usuarios = Deserializador.deserializar(Usuario.usuarios, "Usuarios")
        
    